### ISSUE TITLE
🐛 BUG: when trying to left join 2 tables that have same name columns - second column is not returned

### ISSUE THREAD

**Issue by AndriiSherman** [cloudflare/workers-sdk#3160]
### Which Cloudflare product(s) does this pertain to?

D1

### What version of `Wrangler` are you using?

2.19.0

### What operating system are you using?

macOS

### Describe the Bug

As per issue [#555](https://github.com/drizzle-team/drizzle-orm/issues/555) we can see, that d1 stopped returning duplicate names from different tables. It was working before and it's how any other driver is doing for `raw` mode

```bash
$ pnpm wrangler d1 execute dzltest --command='select "users"."id", "posts"."id", "posts"."ownerId", "posts"."title", "posts"."content" from "users" left join "posts" on "posts"."ownerId" = "users"."id";'

┌────┬─────────┬───────┬───────────┐
│ id │ ownerId │ title │ content   │
├────┼─────────┼───────┼───────────┤
│ 2  │ 1       │ test  │ 111111111 │
├────┼─────────┼───────┼───────────┤
│ 3  │ 1       │ ueoau │ ueueu     │
├────┼─────────┼───────┼───────────┤
│ 1  │ 1       │ xxx   │ yyy       │
└────┴─────────┴───────┴───────────┘
```

But should return `id` for user and `id` for posts as well
Expected
```bash
┌────┬────┬─────────┬───────┬───────────┐
│ id │ id │ ownerId │ title │ content   │
├────┼────┼─────────┼───────┼───────────┤
│ 2  │ 1  │    1    │ test  │ 111111111 │
├────┼────┼─────────┼───────┼───────────┤
│ 3  │ 2  │    1    │ ueoau │ ueueu     │
├────┼────┼─────────┼───────┼───────────┤
│ 1  │ 3  │    1    │ xxx   │ yyy       │
└────┴────┴─────────┴───────┴───────────┘
```

**Comment by JacobMGEvans** [comment]
👋🏻 Greetings,
I appreciate the report. This is the repo and issues for Wrangler (Cloudflare Worker CLI) Pages, & Other SDK tooling, this seems D1 specific.

Feel free to check out the Cloudflare Discord where there is a channel for D1 as well: https://discord.com/invite/cloudflaredev

To report issues related to D1 specifically:
https://community.cloudflare.com/c/developers/d1/85/l/latest

Cc: @rozenmd

**Comment by rozenmd** [comment]
@AndriiSherman could I bother you for a schema/db sql file to reproduce this?

**Comment by xdivby0** [comment]
Although I am not Andrii, I think I have the same problem and here's a minimal example schema that has the issue (pretty much any schema that you can join where both have one shared field name (here it's "id"):
```ts
import type { InferModel } from "drizzle-orm";
import {
  integer, primaryKey,
  sqliteTable,
  text
} from "drizzle-orm/sqlite-core";

export const users = sqliteTable(
  "users",
  {
    id: integer("id").primaryKey({
      autoIncrement: true
    }),
    email: text("email").notNull(),
  }
);

export const subscriptions = sqliteTable(
  "subscriptions",
  {
    id: integer("id").primaryKey({ autoIncrement: true }),
    user: integer("user").notNull().references(() => users.id, {
      onDelete: "cascade"
    })
  }
);

```

**Comment by rozenmd** [comment]
We've traced this back to workerd, and will be discussing possible solutions here: https://github.com/cloudflare/workerd/pull/696

Keep in mind @AndriiSherman that a temporary fix on Drizzle's side is still possible by using `.all()` and manipulating the data into the array of arrays format yourselves.

**Comment by itsmatteomanf** [comment]
> Keep in mind @AndriiSherman that a temporary fix on Drizzle's side is still possible by using .all() and manipulating the data into the array of arrays format yourselves.

I’m pretty sure it behaves wrongly even with this option. I’m gonna need to check, but how could it return multiple identical keys in an array of objects?

Normally SQL databases return the table name in the joined table’s column names. Or even in both.

Do you want me to check, even if it’s impossible to return multiple keys of the same name, @rozenmd?

**Comment by marina-chibizova** [comment]
We have the same problem, does anything block fix for this bug?

**Comment by elithrar** [comment]
We are working on this. Current priority is larger databases, which is the biggest ask from most users.

**Comment by marina-chibizova** [comment]
Thanks for a prompt response @elithrar!

**Comment by jwhits** [comment]
> We are working on this. Current priority is larger databases, which is the biggest ask from most users.

Is there any update on resolving this issue?

**Comment by geelen** [comment]
FYI, this will be fixed D1 once https://github.com/cloudflare/workerd/pull/1586 is merged and rolls out

It will only fix the results of `.raw()`, `.all()` will still have later columns with the same name overwriting earlier ones.

**Comment by geelen** [comment]
This is now released. `.raw()` returns the full results set regardless of column name clashes. Sadly Wrangler still uses `.all()` so this won't be visible except from inside the Worker API, so I'll leave this ticket open for now.

**Comment by elithrar** [comment]
Changelog: https://developers.cloudflare.com/d1/platform/changelog/#2024-02-13

cc @AndriiSherman on the Drizzle team.

**Comment by AndriiSherman** [comment]
> Changelog: https://developers.cloudflare.com/d1/platform/changelog/#2024-02-13
> 
> cc @AndriiSherman on the Drizzle team.

Has it been released already? I would like to run some tests, can't find this version for types on npm

**Comment by dankochetov** [comment]
@rozenmd I've tested this with latest types version (`4.20240320.1`), and the `.raw()` method still doesn't return multiple columns with the same name. This change shouldn't require any actions from our side, since we don't bundle `@cloudflare/workers-types` with Drizzle, it's used as a peer dependency, so it should be enough for users to just update their dependency version. However, it doesn't seem to be the case.

Here's a reproduction repo: https://github.com/dankochetov/drizzle-d1-bug
I've added a raw D1 call to make sure it wasn't Drizzle's fault, and observed the same behavior. As you could see from the logs, the columns list in the response don't match the columns list in the query.

![image](https://github.com/cloudflare/workers-sdk/assets/33990502/d0eb5ac0-4c82-4c9b-91ff-5e015c55ba8c)
![image](https://github.com/cloudflare/workers-sdk/assets/33990502/01e439c9-b0ac-438d-baf6-78c9dfc68b49)

**Comment by geelen** [comment]
Hi @dankochetov, that looks like a bug in local mode (probably related to `getPlatformProxy()`, though I'll need to dig in and figure out exactly what).

The `.raw()` behaviour has been fixed in production though, I've just converted your repro to actually publish a worker and I get the expected output:

![image](https://github.com/cloudflare/workers-sdk/assets/23264/c85e6aba-465e-4eda-8035-36305b95c52d)

I'll take a look at the local behaviour this week, apologies for the oversight.

**Comment by stefanmaric** [comment]
> has been fixed in production though

@geelen I'm using the latest wrangler `3.37.0` to deploy my app with `compatibility_date = "2024-03-25"` and I'm still facing this issue.

Is there anything special I need to do to get the new version?

**Comment by stefanmaric** [comment]
Running my query from the D1 Dashboard console UI or wrangler D1 CLI get me the same, buggy result. It collapses columns with the same name from joined tables into a single one, the last joined table seemingly taking precedence.

**Comment by F0rce** [comment]
Are there any updates on this?

**Comment by kossnocorp** [comment]
I added a test case for the problem: https://github.com/cloudflare/workers-sdk/pull/5917

I went down a rabbit hole chasing this bug through `workerd` (where I also added tests, but it passed), and now, Workers SDK.

Hopefully someone from the maintainers will notice it and help me fix the problem. Any tips will be appriciated.

**Comment by kossnocorp** [comment]
I found the root cause of the problem and fixed it: https://github.com/cloudflare/workers-sdk/pull/5917

Hopefully, the PR will be merged and shipped soon.

**Comment by rozenmd** [comment]
@kossnocorp - that PR specifically fixes Workers when run locally, not the execute command. I'll have a follow-up PR for the execute command shortly.

**Comment by serhii-chernenko** [comment]
I have the issue as well. There's an example:
https://github.com/drizzle-team/drizzle-orm/issues/555#issuecomment-2888926399
